diff --git a/src/py/libcamera/meson.build b/src/py/libcamera/meson.build
index b0bd291f..2c725a45 100644
--- a/src/py/libcamera/meson.build
+++ b/src/py/libcamera/meson.build
@@ -55,7 +55,9 @@ pycamera_args = [
     '-fvisibility=hidden',
     '-Wno-shadow',
     '-DPYBIND11_USE_SMART_HOLDER_AS_DEFAULT',
+    '-DLIBCAMERA_BASE_PRIVATE',
 ]
+subdir_done()
 
 if meson.is_cross_build()
     # Meson doesn't correctly support cross-compilation through the python
